velox/functions/prestosql/tests/ArgGeneratorTest.cpp

```diff
@@ -0,0 +1,153 @@
+      exec::FunctionSignatureBuilder()
+          .integerVariable("a_scale")
+          .integerVariable("b_scale")
+          .integerVariable("a_precision")
+          .integerVariable("b_precision")
+          .integerVariable(
+              "r_precision",
+              "min(38, max({a_precision} - {a_scale}, {b_precision} - {b_scale}) + max({a_scale}, {b_scale}) + 1)")
+          .integerVariable("r_scale", "max({a_scale}, {b_scale})")
+          .returnType("decimal(r_precision, r_scale)")
+          .argumentType("decimal(a_precision, a_scale)")
+          .argumentType("decimal(b_precision, b_scale)")
+          .build();
+
+  const auto generator = std::make_shared<exec::test::PlusMinusArgGenerator>();
+  for (auto returnType :
+       {DECIMAL(10, 2),
+        DECIMAL(32, 6),
+        DECIMAL(38, 20),
+        DECIMAL(38, 38),
+        DECIMAL(38, 0)}) {
+    fuzzer::test::assertReturnType(generator, *signature, returnType);
+  }
+  fuzzer::test::assertEmptyArgs(generator, *signature, DECIMAL(18, 18));
+}
+
+TEST_F(ArgGeneratorTest, multiply) {
+  auto signature =
+      exec::FunctionSignatureBuilder()
+          .integerVariable("a_scale")
+          .integerVariable("b_scale")
+          .integerVariable("a_precision")
+          .integerVariable("b_precision")
+          .integerVariable(
+              "r_precision", "min(38, {a_precision} + {b_precision})")
+          .integerVariable("r_scale", "{a_scale} + {b_scale}")
+          .returnType("decimal(r_precision, r_scale)")
+          .argumentType("decimal(a_precision, a_scale)")
+          .argumentType("decimal(b_precision, b_scale)")
+          .build();
+
+  const auto generator = std::make_shared<exec::test::MultiplyArgGenerator>();
+  for (auto returnType :
+       {DECIMAL(10, 2),
+        DECIMAL(18, 18),
+        DECIMAL(32, 6),
+        DECIMAL(38, 20),
+        DECIMAL(38, 38),
+        DECIMAL(38, 0)}) {
+    fuzzer::test::assertReturnType(generator, *signature, returnType);
+  }
+}
+
+TEST_F(ArgGeneratorTest, divide) {
+  auto signature =
+      exec::FunctionSignatureBuilder()
+          .integerVariable("a_scale")
+          .integerVariable("b_scale")
+          .integerVariable("a_precision")
+          .integerVariable("b_precision")
+          .integerVariable(
+              "r_precision",
+              "min(38, {a_precision} + {b_scale} + max(0, {b_scale} - {a_scale}))")
+          .integerVariable("r_scale", "max({a_scale}, {b_scale})")
+          .returnType("decimal(r_precision, r_scale)")
+          .argumentType("decimal(a_precision, a_scale)")
+          .argumentType("decimal(b_precision, b_scale)")
+          .build();
+
+  const auto generator = std::make_shared<exec::test::DivideArgGenerator>();
+  for (auto returnType :
+       {DECIMAL(10, 2),
+        DECIMAL(18, 18),
+        DECIMAL(32, 6),
+        DECIMAL(38, 20),
+        DECIMAL(38, 38),
+        DECIMAL(38, 0)}) {
+    fuzzer::test::assertReturnType(generator, *signature, returnType);
```

Comment:
nit: it might be easier to read if this is written without a loop

```
fuzzer::test::assertReturnType(generator, *signature, DECIMAL(10, 2));
fuzzer::test::assertReturnType(generator, *signature, DECIMAL(18, 18));
...
```

Reply:
Updated. Thanks.